clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -371,8 +371,7 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         } catch (UnsupportedVersionException e) {
             // If the version is not supported, skip sending the request over the wire.
             // Instead, simply add it to the local queue of aborted requests.
-            log.debug("Version mismatch when attempting to send {} to {}",
-                    clientRequest.toString(), clientRequest.destination(), e);
+            log.debug("Version mismatch when attempting to send {} to {}", clientRequest, clientRequest.destination(), e);
```

Comment:
OK, I replaced `clientRequest` with the request builder. That would have the information missing from `UnsupportedVersionException` (the latter would have the api key and version that was requested).